internal/k8s/client.go

```diff
@@ -301,31 +303,57 @@ func (c *Client) ListServices(ctx context.Context, namespace string, options met
 }
 
 func (c *Client) ExecInPodWithStderr(ctx context.Context, namespace, pod, container string, command []string) (bytes.Buffer, bytes.Buffer, error) {
```

Comment:
I think the code added to this function is redundant, but I hope the feedback I typed up is helpful. This function should simply wrap `c.execInPod`. (basically, replace this function with the contents of `ExecInPodWithTTY` and remove said function) No channels/timeouts required, just a simple wrapper. 

`c.execInPod` should return on its own when the command exits (whether naturally or after a context cancelation). We should never proceed without waiting for the command to finish. If it doesn't finish, that's a fatal error and cilium-cli should terminate. A 30-second timeout would work, returning a timeout sentinel so the test harness can `Fatalf()` accordingly.

---

Some feedback I originally wrote up for this function:
- It's hard to understand what's going on with the nested selects, looks like a source of bugs to me.
- `ctx` is passed down to `c.execInPod`, so waiting for `ctx.Done()` in an enclosing function is a bit pointless. The callee should return ASAP when the context it takes is closed.
- Having a `CloseTimeout` will leak the goroutine calling `c.execInPod`, we might as well fire & forget goroutines instead of implementing cancelation then. :slightly_smiling_face: The cancelation allows us to notify the program so it can return earlier than expected, but we should _always_ wait for it to exit.
- `time.After()` allocates a timer internally, which is not garbage collected until it fires. (see https://medium.com/@oboturov/golang-time-after-is-not-garbage-collected-4cbc94740082) Not a big issue here, but it's best to avoid using it this way in general.

Reply:
I don't know that there would be a guarantee that the remote execution exits on Ctrl-C / Ctrl-D. IMO it would be prudent to not hang in that case so that a proper error can be reported.

Reply:
Leaking goroutines can be covered by exiting from the cilium-cli; leaking remote executions in pods is a bit more serious. As long as they are in the test namespace they will be cleaned up when the test namespace is deleted. While debugging the monitoring I saw leaked remote executions in cilium pods - maybe one more reason to avoid it, even if it usually would terminate properly.

Reply:
Cleaned up.